kotsadm/pkg/handlers/login.go

```diff
@@ -340,18 +340,31 @@ func (h *Handler) OIDCLoginCallback(w http.ResponseWriter, r *http.Request) {
 	}
 
 	expire := time.Now().Add(30 * time.Minute)
-	cookie := http.Cookie{
+
+	// token cookie
+	tokenCookie := http.Cookie{
 		Name:    "token",
 		Value:   responseToken,
 		Expires: expire,
 		Path:    "/",
 	}
+	if strings.HasPrefix(redirectURL, "https") {
```

Comment:
should probably be "https://".   or better yet, `url.Parse` and check `u.Scheme == "https"`